### PR TITLE
Run gofmt -s and gosimple

### PR DESCRIPTION
**Pull request by lucas-clemente**

### CODE REVIEW

File: stream.go [hunk 1]
```diff
@@ -134,7 +134,7 @@ func (s *stream) Read(p []byte) (int, error) {
 			} else {
 				select {
 				case <-s.readChan:
-				case <-time.After(deadline.Sub(time.Now())):
+				case <-time.After(time.Until(deadline)):
```
Comment (marten-seemann) [inline stream.go]: This isn't really easy to read.

Comment (lucas-clemente) [inline stream.go]: True, but I think it's still easier than the old version.

Comment (marten-seemann) [inline stream.go]: The alternative would be `time.NewTimer(deadline).C`. I'm fine with either.

Comment (lucas-clemente) [inline stream.go]: That would cause more garbage, so I'd keep it as is.

**Pull request closed without merging**